src/rattler_build_conda_compat/conditional_list.py

```diff
@@ -0,0 +1,56 @@
+from typing import Any, Callable, Generator, Generic, Optional, TypeVar, Union
+
+T = TypeVar("T")
+
+
+class IfStatement(Generic[T]):
+    if_: Any
+    then: T | list[T]
+    else_: T | list[T] | None
+
+
+ConditionalList = Union[T, "IfStatement[T]", list[Union[T, "IfStatement[T]"]]]
+
+
+def visit_conditional_list(
+    value: ConditionalList[T], evaluator: Optional[Callable[[Any], bool]] = None
```

Comment:
```suggestion
    value: ConditionalList[T], evaluator: Callable[[Any], bool] | None = None
```

Personally, now that we have a nice union syntax, I try to avoid the confusing `Optional`